### DISCUSSION TITLE
Can renovate create auto feature branch and auto merge in main.

### DISCUSSION THREAD

**Discussion by SORABH13** [renovatebot/renovate#16744] [Request Help]
Hi Team,

I am working in renovate in gitlab as self hosted environment. My doubt about renovate is that Whenever repo scan from renovate it will auto create "feature" branch and automerge first PR "renovate configure" in feature branch. Along with renovate scan repo and automerge all dependency package update PR in "feature" branch automatically. Now at least renovate create a single PR for merge "feature" branch into "main". I just want show only one pull request to my team instated of multiple pr. Can any one help me out this issue???

**Comment by secustor** [comment]
What you are looking for is [package grouping](https://docs.renovatebot.com/noise-reduction/#package-grouping)

**Reply by SORABH13** [reply]
No, My Questions are:

1. Can Renovate create "a copy of main branch as feature" branch?
2. Can Renovate automatically merge "renovate.json" file in "feature" branch?
3. Can Renovate automatically scan "feature" branch and after scan dependency automerge with "feature" branch.?
4. Finally all dependency update and added "renovate.json" file in "feature" branch. After that Can renovate create single PR for merge "feature" branch into main branch?

I am looking solution for:

step1: Renovate bot scan repo & automatically create a "feature" branch to add renovate.json file.
Step2. Renovate only scan that "feature" branch for dependency scan instead of scan directly "main" branch. 
Step3: After scan renovate all new dependency update PR automerge with "feature" branch. 
Step4: At the end renovate automatically create a PR for merge "feature" into "main" branch. So that I can only get a single pr which for merge feature to main. 

Can you please help on this. It would really helpful for me.

**Reply by PhilipAbed** [reply]
well you can use [baseBranches](https://docs.renovatebot.com/configuration-options/#basebranches)
then it can [automerge](https://docs.renovatebot.com/configuration-options/#automerge) into that feature branch using the config option.

it will help you do step 1 + 2 + 3 on your other branch.

But Step 4 im afraid you will need to do that in your script/pipeline with git commands

**Reply by SORABH13** [reply]
Thank You so much for reply. Its useful for me..

Have one more question. Is there any way from that renovate in first scan of repo only create branch and don't create pull request?

For example first time renovate create one feature branch and also create one PR "Configure Renovate". So I only want branch and don't want pull request. is there any way to stop PR of "Configure Renovate"?

![image](https://user-images.githubusercontent.com/92920790/180977695-2b1d9460-9574-4d02-b951-d24a898acfea.png)

**Reply by rarkins** [reply]
No, but why do you want to do that?

You can skip onboarding altogether, for example.

**Reply by SORABH13** [reply]
Because whenever I have scan group repo it will create multiple configure PR for every repo. That's why I am looking for that it will create single group configure PR only. 
![image](https://user-images.githubusercontent.com/92920790/181182214-7cdbaa08-bdde-48ff-9849-7b9ccb9dfe76.png)

**Reply by rarkins** [reply]
What does "scan group repo" mean?

**Reply by SORABH13** [reply]
Mean, for example I make Renovate Project group in gitlab and under this I have puted may other repo. Then scan from renovate and it's generating multiple configure request pr for every repo. I want only single PR.

This my renovate config.js file snapshot..
![image](https://user-images.githubusercontent.com/92920790/181186044-f1f65072-c800-4092-a225-bfaeffacedff.png)

**Reply by rarkins** [reply]
If you want a config file per repo then you need an onboarding MR per repo. If you don't want config then you can turn off the requirement as well as onboarding

**Reply by PhilipAbed** [reply]
why are there 3 different users on those 3 prs?

**Reply by SORABH13** [reply]
Hi Team, First time Renovate add "renovate.json" file in repo but after some I have done some changes in "config.js" file but when I have scanned my repo but renovate doesn't create new pull request to update old renovate.json file in my repo. Please suggest solution for this Like whenever I have done changes in config.js file renovate automatically create new PR for update old renovate.json file.

**Reply by rarkins** [reply]
Instead, you should create a central "preset" and have each of your repos extend that so that changes are automatically flowed through without needing PRs. Read https://docs.renovatebot.com/key-concepts/presets/

**Reply by SORABH13** [reply]
If I tried with preset as suggest in document but it's not working showing issue dashboard  with error "Cannot find preset's package (xyz/renovate/renovate-config)".

Do we have any other option like anytime update in config.js and it will create pr for update older renovate.json. 

As you suggested I have tried but its showing error.  Can you please share any example of code and anything which can help me to configure preset or other solution to solve my issue.

**Reply by rarkins** [reply]
I have no further suggestions and you have not provided any logs in order for us to help you find your mistake

**Reply by PhilipAbed** [reply]
Renovate doesnt create more than 1 onboarding PR. When u change your config.js u also are changing something .. probably the user 
Which will cause renovate to not find the old pr and update it. 
Renovate doesnt support multiple users. 
If u want multiple users .. add onboarding:false to the other users  but u might find other wrong things happening too so dont use multiple users
